src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1288,12 +1288,9 @@ private sealed class LoaderHolder
                 ch.Assert(h.Loader == null || h.Loader is ICommandLineComponentFactory);
                 var loader = h.Loader as ICommandLineComponentFactory;
 
-                if (loader == null || string.IsNullOrWhiteSpace(loader.Name))
-                    goto LDone;
-
-                // Make sure the loader binds to us.
-                var info = host.ComponentCatalog.GetLoadableClassInfo<SignatureDataLoader>(loader.Name);
-                if (info.Type != typeof(ILegacyDataLoader) || info.ArgType != typeof(Options))
+                // Make sure that the schema is described using either the syntax TextLoader{<settings>} or the syntax Text{<settings>},
+                // where "settings" is a string that can be parsed by CmdParser into an object of type TextLoader.Options.
+                if (loader == null || string.IsNullOrWhiteSpace(loader.Name) || (loader.Name != LoaderSignature && loader.Name != "Text"))
```

Comment:
>Name [](start = 88, length = 4)

The previous check was to verify that `loader.Name` is one of the names defined in `TextLoader`'s `LoadableClassAttribute`. Since now we cannot rely on this attribute being in the `ComponentCatalog`, we explicitly require `loader.Name` to be one of the load names of `TextLoader`.